app/models/claim/base_claim.rb

```diff
@@ -389,25 +393,18 @@ def disabled_for_transition?
       disable_for_state_transition.present?
     end
 
-    def next_step!
-      self.form_step = submission_stages[current_step_index + 1]
-      until current_step_required?
-        break unless form_step
-        self.form_step = submission_stages[current_step_index + 1]
-      end
-      form_step
+    def submission_stages
+      @submission_stages ||= StageCollection.new(self.class::SUBMISSION_STAGES, self)
     end
 
-    def current_step_index
-      submission_stages.index(current_step)
+    def submission_current_flow
+      return submission_stages if from_api?
+      submission_stages.path_until(form_step)
     end
 
-    def current_step_required?
```

Comment:
🏌️ Got rid of this completely, as any conditions are now managed through the submission _stage manager_.